// @firebolt-js/manage-sdk/dist/lib/HDMIInput/defaults.mjs
/*
 * Copyright 2021 Comcast Cable Communications Management, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
import MockProps from '../Prop/MockProps.mjs'

export default {
  ports: [
    {
      port: 'HDMI1',
      connected: true,
      signal: 'stable',
      arcCapable: true,
      arcConnected: true,
      edidVersion: '2.0',
      autoLowLatencyModeCapable: true,
      autoLowLatencyModeSignalled: true,
    },
  ],
  port: {
    port: 'HDMI1',
    connected: true,
    signal: 'stable',
    arcCapable: true,
    arcConnected: true,
    edidVersion: '2.0',
    autoLowLatencyModeCapable: true,
    autoLowLatencyModeSignalled: true,
  },
  open: null,
  close: null,
  lowLatencyMode: function (params) {
    return MockProps.mock(
      'HDMIInput',
      'lowLatencyMode',
      params,
      undefined,
      0,
      true,
    )
  },
  autoLowLatencyModeCapable: function (params) {
    return MockProps.mock(
      'HDMIInput',
      'autoLowLatencyModeCapable',
      params,
      undefined,
      1,
      true,
    )
  },
  edidVersion: function (params) {
    return MockProps.mock(
      'HDMIInput',
      'edidVersion',
      params,
      undefined,
      1,
      '2.0',
    )
  },
  setLowLatencyMode: function (params) {
    const callbackOrValue = params.value
    delete params.value
    return MockProps.mock(
      'HDMIInput',
      'lowLatencyMode',
      params,
      callbackOrValue,
      0,
      null,
    )
  },
  setAutoLowLatencyModeCapable: function (params) {
    const callbackOrValue = params.value
    delete params.value
    return MockProps.mock(
      'HDMIInput',
      'autoLowLatencyModeCapable',
      params,
      callbackOrValue,
      0,
      null,
    )
  },
  setEdidVersion: function (params) {
    const callbackOrValue = params.value
    delete params.value
    return MockProps.mock(
      'HDMIInput',
      'edidVersion',
      params,
      callbackOrValue,
      0,
      null,
    )
  },
}
