// @firebolt-js/manage-sdk/dist/lib/firebolt-manage.mjs
/*
 * Copyright 2021 Comcast Cable Communications Management, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { setMockResponses } from './Transport/MockTransport.mjs'

import { default as _Localization } from './Localization/defaults.mjs'
import { default as _Metrics } from './Metrics/defaults.mjs'
import { default as _Wifi } from './Wifi/defaults.mjs'
import { default as _ClosedCaptions } from './ClosedCaptions/defaults.mjs'
import { default as _AudioDescriptions } from './AudioDescriptions/defaults.mjs'
import { default as _VoiceGuidance } from './VoiceGuidance/defaults.mjs'
import { default as _Device } from './Device/defaults.mjs'
import { default as _UserGrants } from './UserGrants/defaults.mjs'
import { default as _Privacy } from './Privacy/defaults.mjs'
import { default as _Advertising } from './Advertising/defaults.mjs'
import { default as _Account } from './Account/defaults.mjs'
import { default as _Keyboard } from './Keyboard/defaults.mjs'
import { default as _AcknowledgeChallenge } from './AcknowledgeChallenge/defaults.mjs'
import { default as _PinChallenge } from './PinChallenge/defaults.mjs'
import { default as _SecureStorage } from './SecureStorage/defaults.mjs'
import { default as _Discovery } from './Discovery/defaults.mjs'
import { default as _HDMIInput } from './HDMIInput/defaults.mjs'

setMockResponses({
  Localization: _Localization,
  Metrics: _Metrics,
  Wifi: _Wifi,
  ClosedCaptions: _ClosedCaptions,
  AudioDescriptions: _AudioDescriptions,
  VoiceGuidance: _VoiceGuidance,
  Device: _Device,
  UserGrants: _UserGrants,
  Privacy: _Privacy,
  Advertising: _Advertising,
  Account: _Account,
  Keyboard: _Keyboard,
  AcknowledgeChallenge: _AcknowledgeChallenge,
  PinChallenge: _PinChallenge,
  SecureStorage: _SecureStorage,
  Discovery: _Discovery,
  HDMIInput: _HDMIInput,
})

export { default as Localization } from './Localization/index.mjs'
export { default as Metrics } from './Metrics/index.mjs'
export { default as Wifi } from './Wifi/index.mjs'
export { default as ClosedCaptions } from './ClosedCaptions/index.mjs'
export { default as AudioDescriptions } from './AudioDescriptions/index.mjs'
export { default as VoiceGuidance } from './VoiceGuidance/index.mjs'
export { default as Device } from './Device/index.mjs'
export { default as UserGrants } from './UserGrants/index.mjs'
export { default as Privacy } from './Privacy/index.mjs'
export { default as Advertising } from './Advertising/index.mjs'
export { default as Account } from './Account/index.mjs'
export { default as Keyboard } from './Keyboard/index.mjs'
export { default as AcknowledgeChallenge } from './AcknowledgeChallenge/index.mjs'
export { default as PinChallenge } from './PinChallenge/index.mjs'
export { default as SecureStorage } from './SecureStorage/index.mjs'
export { default as Discovery } from './Discovery/index.mjs'
export { default as HDMIInput } from './HDMIInput/index.mjs'
export { default as Log } from './Log/index.mjs'
export { default as Events } from './Events/index.mjs'
export { default as Settings } from './Settings/index.mjs'
